src/v/compression/stream_zstd.cc

```diff
@@ -46,18 +67,16 @@ stream_zstd::zstd_compress_ctx& stream_zstd::compressor() {
     return _compress;
 }
 
-void stream_zstd::reset_decompressor() {
-    _decompress.reset(ZSTD_createDCtx());
-    if (!_decompress) {
-        throw std::bad_alloc{};
-    }
-}
-
-stream_zstd::zstd_decompress_ctx& stream_zstd::decompressor() {
-    if (!_decompress) {
-        reset_decompressor();
+ZSTD_DCtx* stream_zstd::decompressor() {
+    if (unlikely(!dctx_workspace)) {
```

Comment:
why not do this inside this function scope 

`static thread local foo = new char[2_MB]();` kind of thing. 

the compiler will generate basically the same assembly

Reply:
this branch is only used for tests. im not really sure what your motivation for the change is